crates/protocol/src/message/quote.rs

```diff
@@ -0,0 +1,140 @@
+use ::serde::{Deserialize, Serialize};
+use chrono::{DateTime, Utc};
+use serde_with::skip_serializing_none;
+use type_safe_id::{DynamicType, TypeSafeId};
+
+use super::{Message, MessageError, MessageKind, MessageMetadata};
+
+pub struct Quote;
+
+impl Quote {
+    pub fn create(
+        from: String,
+        to: String,
+        exchange_id: TypeSafeId<DynamicType>,
+        data: QuoteData,
+    ) -> Result<Message<QuoteData>, MessageError> {
+        let metadata = MessageMetadata {
+            from,
+            to,
+            kind: MessageKind::Quote,
+            id: MessageKind::Quote.typesafe_id()?,
+            exchange_id,
+            created_at: Utc::now(),
+        };
+
+        Ok(Message {
+            metadata,
+            data,
+            signature: None,
+        })
+    }
+}
+
+/// A struct representing the data contained within the [`Message`] for a [`Quote`].
+///
+/// See [Quote](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#quote) for more
+/// information.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[skip_serializing_none]
+#[serde(rename_all = "camelCase")]
+pub struct QuoteData {
+    /// When this quote expires. Expressed as ISO8601
+    pub expires_at: DateTime<Utc>,
+    /// the amount of payin currency that the PFI will receive
+    pub payin: QuoteDetails,
+    /// the amount of payout currency that Alice will receive
+    pub payout: QuoteDetails,
+    /// Object that describes how to pay the PFI, and how to get paid by the PFI (e.g. BTC address, payment link)
+    pub payment_instructions: Option<PaymentInstructions>,
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[skip_serializing_none]
+#[serde(rename_all = "camelCase")]
+pub struct QuoteDetails {
+    /// ISO 3166 currency code string
+    pub currency_code: String,
+    /// The amount of currency expressed in the smallest respective unit
+    pub amount_subunits: String,
```

Comment:
Currently, the spec defines them as strings here: https://github.com/TBD54566975/tbdex/tree/main/specs/protocol
We also define them as strings in [Javascript](https://github.com/TBD54566975/tbdex-js/blob/5631d324bbacbe3d52236f64eb2110bafb00c55b/packages/protocol/src/types.ts#L229) & [Kotlin](https://github.com/TBD54566975/tbdex-kt/blob/6ab8c84d3c62fbe11b45fb36e329e5ecbac4b0e2/protocol/src/main/kotlin/tbdex/sdk/protocol/models/MessageData.kt#L48)

Definitely open to changing, but we might want to modify the spec if this is something we want to do. For now, should be fine though because it meets the existing spec!

Reply:
That's a good point. One thing I do want to mention though is that tbDEX messages/resources will need to traverse the language boundary from Rust -> other foreign languages. Strings work really well in this regard, third-party crate types not so much.

It's definitely going to be some tradeoffs, but I think we can find a happy medium here. Looking forward to discussing this more!

Reply:
@andresuribe87 There's ongoing discussion about currency amount representation in #199. For now I'm going to stick with plain ol strings until we make a decision in the spec.